substrate/client/service/src/builder.rs

```diff
@@ -372,6 +373,19 @@ pub struct SpawnTasksParams<'a, TBl: BlockT, TCl, TExPool, TRpc, Backend> {
 	pub telemetry: Option<&'a mut Telemetry>,
 }
 
+// Wrapper for HTTP and WS servers that makes sure they are properly shut down.
+mod waiting {
```

Comment:
From a quick glance, I don't see it is used anywhere? Is it acutally needed?

Reply:
good catch, added a suggestion for it